composer/trainer/trainer.py

```diff
@@ -1652,15 +1663,21 @@ def _train_batch(self, use_grad_scaling: bool):
                             total_loss = self.state.scaler.step(
                                 optimizer, closure=lambda **kwargs: self._train_microbatches(microbatches, **kwargs))
                         else:
-                            total_loss = optimizer.step(
-                                closure=lambda **kwargs: self._train_microbatches(microbatches, **kwargs).item())
+                            if type(self._device) == DeviceTPU:
+                                total_loss = xm.optimizer_step(optimizer)
+                            else:
+                                total_loss = optimizer.step(
+                                    closure=lambda **kwargs: self._train_microbatches(microbatches, **kwargs).item())
                 else:
                     total_loss = self._train_microbatches(microbatches)
                     for optimizer in self.state.optimizers:
-                        if use_grad_scaling:
-                            self.state.scaler.step(optimizer)
+                        if type(self._device) == DeviceTPU:
+                            xm.optimizer_step(optimizer)
```

Comment:
How do we handle grad scaling if using tpu? I saw we create a XLA scaler earlier but I don't see that being stepped here?

Reply:
fixed